apis/duck/cached.go

```diff
@@ -38,35 +38,51 @@ var _ InformerFactory = (*CachedInformerFactory)(nil)
 // Get implements InformerFactory.
 func (cif *CachedInformerFactory) Get(gvr schema.GroupVersionResource) (cache.SharedIndexInformer, cache.GenericLister, error) {
 	cif.m.Lock()
+
 	if cif.cache == nil {
-		cif.cache = make(map[schema.GroupVersionResource]*result)
+		cif.cache = make(map[schema.GroupVersionResource]*informerCache)
 	}
-	elt, ok := cif.cache[gvr]
+
+	ic, ok := cif.cache[gvr]
 	if !ok {
-		elt = &result{}
-		elt.init = func() {
-			elt.inf, elt.lister, elt.err = cif.Delegate.Get(gvr)
+		ic = &informerCache{}
+		ic.init = func() {
+			ic.Lock()
```

Comment:
You want to do a two step check here (https://en.wikipedia.org/wiki/Double-checked_locking)
Since in theory more than 1 goroutine may pass `!initialized` check and end up here initializing twice.

Reply:
You're right, I omitted it to keep things simple but will add the extra check.

Reply:
[Addressed](https://github.com/knative/pkg/compare/27fcb36f1cec5f30c95682459ef3e3314183d2df..ab2ea61156d781737e67f63f3891c479ea56210e)